### ISSUE TITLE
Missing Kustomize Dependency for `make install`

### ISSUE THREAD

**Issue by kris-nova** [kubernetes-sigs/kubebuilder#1371]
Fresh linux system, trying to go through the getting started
```
[nova@nova kapad]$ make install
/home/nova/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
kustomize build config/crd | kubectl apply -f -
/bin/sh: kustomize: command not found
error: no objects passed to apply
make: *** [Makefile:30: install] Error 1
```

Either docs that mention this, a check in `make install`, a better error message with a path forward, or including `kustomize` in the `build/bin/` directory of a release

/kind bug

**Comment by droot** [comment]
[Kubebuilder book](https://book.kubebuilder.io/quick-start.html#prerequisites) mentions kustomize in prerequisites section in quick-start-guide.

Though better error message pointing to kustomize install page would be great help here.

**Comment by bharathi-tenneti** [comment]
@droot 
Did you mean the check to be done in [this](https://github.com/kubernetes-sigs/kubebuilder/blob/master/Makefile) file?

**Comment by camilamacedo86** [comment]
HI @bharathi-tenneti, 

It is not regarded the makefile of the kubebuilder project. The makefile in the kubebuilder project are targets which are helpful for who will contribute with the project. 

In this scenario,  the user says `Fresh linux system, trying to go through the getting started` which means that he is following the the [Quick Started](https://book.kubebuilder.io/quick-start.html) and the issue is faced in the step:

<img width="897" alt="Screenshot 2020-03-09 at 19 07 23" src="https://user-images.githubusercontent.com/7708031/76248431-3af4c880-6239-11ea-9c35-fda86f02843b.png">

So, what @droot is suggestion here as solution is: 

Improve the error face in the this scenario in order to allow users to know that the prerequisite kustomize is not installed. Also, I would recommend we solve it by checking the pre-requirement is install as we do or the Go version see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/cmd/internal/go_version.go

**Comment by Adirio** [comment]
I agree with @camilamacedo86 that checking the dependency could be done. I don't think there is any other easy way to improve the error message, as it is a OS error not one we are printing, so we need to check for it before running any kustomize command.

**Comment by bharathi-tenneti** [comment]
@Adirio I agree that checking dependency can be done for kustomize.
However, if we go down the path, Should we go ahead an check for other [pre-reqs](https://book.kubebuilder.io/quick-start.html#prerequisites) as well? 
Do we want to do that ?

**Comment by camilamacedo86** [comment]
Hi @bharathi-tenneti, 

To solve the scenario raise here (close this issue) I would just to do the check for the kustomzie one since it will raise an issue that is not intuitive at all. Also, I do not think that this would be the case of docker and/or kubetcl as well.

However, after doing this one if you wish I'd recommend do a test without docker and see if is clear or not the msg and then check if it is required or not.

**Comment by camilamacedo86** [comment]
@bharathi-tenneti, 

Feel free to do. the. other checks if you see that would be nice.  To close this one IHMO doing it for kustomize only would be enough.

**Comment by Adirio** [comment]
I was working on this, but don't have the PR ready yet.
I also wanted to move this check out of `kubebuilder init` just to any `kubebuilder` command so that when you git clone an already init project you still get this check done for you.

**Comment by dmizelle** [comment]
Eek. I didn't realize this was assigned to someone else (its 2AM here and I'm unable to sleep)

Didn't mean to step on someone else that was interested in doing this. I ran into this issue tonight as I just rely on `kustomize` built into `kubectl` usually. If you've already done this (or are working on it) just feel free to close my PR. :)

**Comment by bharathi-tenneti** [comment]
@dmizelle no worries, I have not got around to it yet, so go ahead.

**Comment by Adirio** [comment]
He made a proposal on #1427 and I made mine in #1428.

**Comment by toonsevrin** [comment]
Once #1430 is merged, kustomize will be installed automatically when not present in PATH.

**Comment by Adirio** [comment]
So we could either install or just check for the dependency. Both aproaches are being used currently: go version is checked while controller-gen is installed if not present. I'm not sure which of them feels better.

@camilamacedo86 @mengqiy @droot @DirectXMan12 what approach do you prefer?

**Comment by camilamacedo86** [comment]
Hi @Adirio, 

I think that go is a pre-requirement that could not be changed. However, doing it by the Makefile allows the user to customize it which shows more appropriate in this case. So, I am inclined to say for we forward with  #1430.

**Comment by toonsevrin** [comment]
I've commented this on #1430, but ideally the dependencies like controller-gen and kustomize should only be installed locally (eg. local to the project). This could be in `${PROJECT_DIR}/bin` for example. 

As they are internal dependencies in my opinion, they should not change the external environment (and ideally not be influenced by it either, all though that would break backwards compatibility).

Anyways, this is perfectionistic, so I would simply go with #1430 as well!

**Comment by Adirio** [comment]
I do agree that having a `bin` folder with local dependencies sounds better than globally installed. But until we opt into thta approach, I would not install things for users, I would just check if they are installed. Basically because we are changing their global system and not just our local directory.

**Comment by camilamacedo86** [comment]
IHMO, the kustomize is a requirement dependence such as the controller-gen and the solution #1430  make it easier for the users and bring better experience, and address the issue raised here and still allowing users customize it according to their needs since the implementation is in the makefile. Then, I cannot see any reason why not move forward with #1430. 

@mengqiy wdyt ?

**Comment by Adirio** [comment]
@droot @mengqiy @DirectXMan12 would you mind answering the poll raised in this https://github.com/kubernetes-sigs/kubebuilder/issues/1371#issuecomment-599045994?

**Comment by camilamacedo86** [comment]
/close

It was solved with https://github.com/kubernetes-sigs/kubebuilder/pull/1430
